### PR TITLE
support ONNX export of XDropout in deberta{,_v2} and sew_d

### PR DESCRIPTION
**Pull request by garymm**
# What does this PR do?

Enables `torch.onnx.export` of the `StableDropout` module in training mode.
In training mode, the `XDropout` torch.autograd.Function is included. This change
adds a `symbolic` function to `XDropout` which produces an ONNX graph that
is equivalent to the `forward` function.

## Before submitting

- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
No. LMK if you want me to open an issue.

- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).

No, LMK if there's some doc that I should update.

- [x] Did you write any new necessary tests?

## Who can review?

@LysandreJik

### CODE REVIEW

**Review by michaelbenayoun — COMMENTED** [review]
LGTM, thanks for adding this!

About the tests, I would suggest you add it to `tests/onnx/test_onnx_v2.py`, something like "test_pytorch_export_in_training_mode".

**Comment by garymm** [comment]
@michaelbenayoun thanks for taking a look. I added a test.
Unfortunately I couldn't use the existing testing style because none of the affected models have an ONNX config (that's tracked by https://github.com/huggingface/transformers/issues/16308).

**Review by prathikr — APPROVED** [review]

File: tests/onnx/test_onnx_v2.py [hunk 1]
```diff
@@ -337,3 +337,46 @@ def test_tensorflow_export_seq2seq_with_past(
...
+
+    @require_torch
+    @pytest.mark.filterwarnings("ignore:.*Dropout.*:UserWarning:torch.onnx.*")  # torch.onnx is spammy.
+    def test_StableDropout(self):
+        """Tests export of StableDropout in training mode."""
+        import os
+
+        import torch
+
+        from transformers.models.deberta import modeling_deberta
```
Comment (michaelbenayoun) [inline tests/onnx/test_onnx_v2.py:353]: I would make those imports at the top of the file under the `if torch_available()`.

File: tests/onnx/test_onnx_v2.py [hunk 2]
```diff
@@ -337,3 +337,46 @@ def test_tensorflow_export_seq2seq_with_past(
...
+
+
+class LayerTestCase(TestCase):
+    """Tests export of specific Layers / Modules."""
+
+    @require_torch
+    @pytest.mark.filterwarnings("ignore:.*Dropout.*:UserWarning:torch.onnx.*")  # torch.onnx is spammy.
+    def test_StableDropout(self):
+        """Tests export of StableDropout in training mode."""
+        import os
```
Comment (michaelbenayoun) [inline tests/onnx/test_onnx_v2.py:349]: Maybe import at the top of the file

Comment (garymm) [reply tests/onnx/test_onnx_v2.py]: Done.

File: tests/onnx/test_onnx_v2.py (hunk 1 above)
Comment (garymm) [reply tests/onnx/test_onnx_v2.py]: Done.

File: tests/onnx/test_onnx_v2.py [hunk 3]
```diff
@@ -337,3 +337,46 @@ def test_tensorflow_export_seq2seq_with_past(
...
+
+        import torch
+
+        from transformers.models.deberta import modeling_deberta
+
+        devnull = open(os.devnull, "wb")
```
[the remark this answers is not in the document]
Comment (garymm) [reply tests/onnx/test_onnx_v2.py:355]: That's only needed because the exported model is being used after export. In this case there's no reason for it, and it would likely run more slowly.

File: tests/onnx/test_onnx_v2.py [hunk 4]
```diff
@@ -337,3 +337,46 @@ def test_tensorflow_export_seq2seq_with_past(
         self, test_name, name, model_name, feature, onnx_config_class_constructor
     ):
         self._onnx_export(test_name, name, model_name, feature, onnx_config_class_constructor)
+
+
+class LayerTestCase(TestCase):
```
[the remark this answers is not in the document]
Comment (garymm) [reply tests/onnx/test_onnx_v2.py]: Not sure what you meant, but I tried to guess.
If you're not happy with this, please suggest an alternative.

**Review by garymm — COMMENTED** [review]
Thanks for the review! @michaelbenayoun PTAL

**Review by prathikr — APPROVED** [review]

**Comment by garymm** [comment]
@michaelbenayoun bump

**Comment by prathikr** [comment]
@garymm any update on this?

**Comment by prathikr** [comment]
@michaelbenayoun do you have any updates on this PR?

**Review by michaelbenayoun — APPROVED** [review]
LGTM

**Comment by LysandreJik** [comment]
Thank you for your contribution

**Review by LysandreJik — APPROVED** [review]
